src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy

```diff
@@ -21,11 +21,29 @@ class CourseInstallDecoderService implements CourseInstallDecoderPort {
     }
 
     @Override
-    void decodeCourseInstallEvent( String event, int schemaId ) {
+    void decodeCourseInstallV1ToV1( String event, int schemaId ) {
+        def schema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        log.info( 'The avro schema for event decoding : ' + schema.toString( true ) )
+
+        decoder.decodeEventWithOldSchema( schema, event )
```

Comment:
Just curious about the name `decodeEventWithOldSchema`. Does this imply that the service needs to know ahead of time what version of an event it is dealing with and use the proper schema?